conjure-java-core/src/integrationInput/java/com/palantir/product/EteBinaryServiceEndpoints.java

```diff
@@ -1,127 +1,229 @@
+            this.runtime = runtime;
+            this.delegate = delegate;
         }
 
-        private class GetOptionalBinaryPresentHandler implements HttpHandler {
-            @Override
-            public void handleRequest(HttpServerExchange exchange) throws IOException {
-                AuthHeader authHeader = Auth.header(exchange);
-                Optional<BinaryResponseBody> result = delegate.getOptionalBinaryPresent(authHeader);
-                if (result.isPresent()) {
-                    BinarySerializers.serialize(result.get(), exchange);
-                } else {
-                    exchange.setStatusCode(StatusCodes.NO_CONTENT);
-                }
-            }
+        @Override
+        public void handleRequest(HttpServerExchange exchange) throws IOException {
+            AuthHeader authHeader = runtime.auth().header(exchange);
+            InputStream body = runtime.serde().deserializeInputStream(exchange);
+            BinaryResponseBody result = delegate.postBinary(authHeader, body);
+            runtime.serde().serialize(result, exchange);
+        }
+
+        @Override
+        public HttpString method() {
+            return Methods.POST;
+        }
+
+        @Override
+        public String template() {
+            return "/binary";
+        }
+
+        @Override
+        public String serviceName() {
+            return "UndertowEteBinaryService";
+        }
+
+        @Override
+        public String name() {
+            return "postBinary";
+        }
+
+        @Override
+        public HttpHandler handler() {
+            return this;
+        }
+    }
+
+    private static final class GetOptionalBinaryPresentEndpoint implements HttpHandler, Endpoint {
+        private final UndertowRuntime runtime;
+
+        private final UndertowEteBinaryService delegate;
+
+        GetOptionalBinaryPresentEndpoint(
+                UndertowRuntime runtime, UndertowEteBinaryService delegate) {
+            this.runtime = runtime;
+            this.delegate = delegate;
         }
 
-        private class GetOptionalBinaryEmptyHandler implements HttpHandler {
-            @Override
-            public void handleRequest(HttpServerExchange exchange) throws IOException {
-                AuthHeader authHeader = Auth.header(exchange);
-                Optional<BinaryResponseBody> result = delegate.getOptionalBinaryEmpty(authHeader);
-                if (result.isPresent()) {
-                    BinarySerializers.serialize(result.get(), exchange);
-                } else {
-                    exchange.setStatusCode(StatusCodes.NO_CONTENT);
-                }
+        @Override
+        public void handleRequest(HttpServerExchange exchange) throws IOException {
+            AuthHeader authHeader = runtime.auth().header(exchange);
+            Optional<BinaryResponseBody> result = delegate.getOptionalBinaryPresent(authHeader);
+            if (result.isPresent()) {
```

Comment:
it might be kind of nice to handle optionals explicitly in the runtime

Reply:
could imagine an explicit SerDe method which accepts `Optional<?>` and sets the NO_CONTENT on the status, for instance.

Reply:
That would be nice, however it would need to support `Optional`, `OptionalInt` `OptionalDouble`, as well as `alias: optional<*`. This would also duplicate our jackson configuration for serializing optional values which is also used for nested fields. I don't think it's the right approach right now, and supporting it would be an additive change we can make later.